x-pack/plugins/upgrade_assistant/public/application/components/tabs/checkup/deprecations/reindex/flyout/warnings_step.tsx

```diff
@@ -68,175 +69,170 @@ interface WarningsConfirmationFlyoutProps {
   advanceNextStep: () => void;
 }
 
-interface WarningsConfirmationFlyoutState {
-  checkedIds: CheckedIds;
-}
-
 /**
  * Displays warning text about destructive changes required to reindex this index. The user
  * must acknowledge each change before being allowed to proceed.
  */
-export class WarningsFlyoutStep extends React.Component<
-  WarningsConfirmationFlyoutProps,
-  WarningsConfirmationFlyoutState
-> {
-  constructor(props: WarningsConfirmationFlyoutProps) {
-    super(props);
-
-    this.state = {
-      checkedIds: props.warnings.reduce((checkedIds, warning) => {
-        checkedIds[idForWarning(warning)] = false;
-        return checkedIds;
-      }, {} as { [id: string]: boolean }),
+export const WarningsFlyoutStep: React.FunctionComponent<WarningsConfirmationFlyoutProps> = ({
+  warnings,
+  renderGlobalCallouts,
+  closeFlyout,
+  advanceNextStep,
+}) => {
+  const [checkedIds, setCheckedIds] = useState<CheckedIds>(
+    warnings.reduce((initialCheckedIds, warning) => {
+      initialCheckedIds[idForWarning(warning)] = false;
+      return initialCheckedIds;
+    }, {} as { [id: string]: boolean })
+  );
+
+  // Do not allow to proceed until all checkboxes are checked.
+  const blockAdvance = Object.values(checkedIds).filter((v) => v).length < warnings.length;
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const optionId = e.target.id;
+    const nextCheckedIds = {
+      ...checkedIds,
+      ...{
+        [optionId]: !checkedIds[optionId],
+      },
     };
-  }
 
-  public render() {
-    const { warnings, closeFlyout, advanceNextStep, renderGlobalCallouts } = this.props;
-    const { checkedIds } = this.state;
+    setCheckedIds(nextCheckedIds);
+  };
+
+  const { docLinks } = useAppContext();
+  const { ELASTIC_WEBSITE_URL } = docLinks;
+  const esDocBasePath = `${ELASTIC_WEBSITE_URL}guide/en/elasticsearch/reference`;
+  const observabilityDocBasePath = `${ELASTIC_WEBSITE_URL}guide/en/observability`;
+
+  // TODO: Revisit warnings returned for 8.0 upgrade; many of these are likely obselete now
```

Comment:
I think most of the doc links in this file will end up being obsolete for the 8.0 upgrade, but leaving as-is for now. This will be addressed once we better understand reindexing requirements for 8.0.